src/metkit/api/metkit_c.h

```diff
@@ -0,0 +1,238 @@
+ * the MetKit as shared library in Python.
+ * @return int Error code
+ */
+int metkit_initialise();
+
+/* ---------------------------------------------------------------------------------------------------------------------
+ * PARSING
+ * --- */
+
+/**
+ * Parse MARS requests into RequestIterator of Request instances. Resulting RequestIterator
+ * must be deallocated with metkit_free_requestiterator
+ * @param str MARS requests
+ * @param requests Allocates RequestIterator object
+ * @return int Error code
+ */
+int metkit_parse_marsrequest(const char* str, metkit_requestiterator_t** requests, bool strict = false);
+
+/* ---------------------------------------------------------------------------------------------------------------------
+ * REQUEST
+ * --- */
+
+/** Allocates new Request object. Must be deallocated with mekit_free_request
+ * @param request new Request instance
+ * @return int Error code
+ */
+int metkit_new_marsrequest(metkit_marsrequest_t** request);
+
+/** Deallocates Request object and associated resources.
+ * @param request Request instance
+ * @return int Error code
+ */
+int metkit_free_marsrequest(const metkit_marsrequest_t* request);
+
+/** Add parameter and values to request
+ * @param request Request instance
+ * @param param parameter name
+ * @param values array of values for parameter
+ * @param numValues number of values
+ * @return int Error code
+ */
+int metkit_marsrequest_add(metkit_marsrequest_t* request, const char* param, const char* values[], int numValues);
+
+/** Add parameter and values to request
+ * @param request Request instance
+ * @param param parameter name
+ * @param values value to add
+ * @return int Error code
+ */
+int metkit_marsrequest_add1(metkit_marsrequest_t* request, const char* param, const char* value);
+
+/** Set verb in Request object
+ * @param request Request instance
+ * @param verb verb to set
+ * @return int Error code
+ */
+int metkit_marsrequest_set_verb(metkit_marsrequest_t* request, const char* verb);
+
+/** Returns the verb in Request object
+ * @param request Request instance
+ * @param verb verb in request
+ * @return int Error code
+ */
+int metkit_marsrequest_verb(const metkit_marsrequest_t* request, const char** verb);
+
+/** Returns whether parameter is in Request object
+ * @param request Request instance
+ * @param param parameter name
+ * @param has whether parameter exists in request
+ * @return int Error code
+ */
+int metkit_marsrequest_has_param(const metkit_marsrequest_t* request, const char* param, bool* has);
+
+/** Returns ParamIterator of parameters in request. Resulting ParamIterator
+ * must be deallocated with metkit_free_paramiterator
+ * @param request Request instance
+ * @param params Allocates ParamIterator object for parameter names in request
+ * @return int Error code
+ */
+int metkit_marsrequest_params(const metkit_marsrequest_t* request, metkit_paramiterator_t** params);
```

Comment:
It might be easier to return the number of params, and then access them by index? Avoid creating/disposing of the paramiterator?

Reply:
I agree.